dbms/src/Flash/Coprocessor/DAGExpressionAnalyzer.cpp

```diff
@@ -332,11 +334,15 @@ void DAGExpressionAnalyzer::buildCommonAggFunc(
     Names arg_names;
     DataTypes arg_types;
     TiDB::TiDBCollators arg_collators;
-    for (Int32 i = 0; i < child_size; ++i)
+
+    // if count(not null const), just remove the argument.
```

Comment:
Can `count(not_null_column)` also be converted to `count()` ?
Or tidb will do this conversion

Reply:
I aggree that we should also convert `count(not null column)` to `count()`, and why not check the `arg_types`? If you check the `arg_types`, you can just check if the type is nullable or not, don't need to care whether it is constant or not.

Reply:
> Can `count(not_null_column)` also be converted to `count()` ?

For `count(not_null_column)`, there is no convertToFullColumn cost, so convert to `count()` may make no sense?
But for `count(not_null_column)`, TiDB will request the not_null_column to scan. If there are delta layer and multiple version in stable layer, we need to scan pk, version, del_tag column to do mvcc filter. In this case, actually we request convert `count(not_null_column)` to `count()` and request pk column to scan instead of not_null_column. But if there is no delta layer and only one version in stable, then there is no need to do mvcc filter, so request pk column maybe heavier than not_null_column.